app/pages/sections/order-items/additional-services/order-item/flat/ondemand/ui-tests/general.ui.test.js

```diff
@@ -54,239 +94,222 @@ const pageSetup = async (setup = defaultPageSetup) => {
+    .expect(await extractInnerText(solutionName)).eql('CHP LTD - BRANSHOLME (B81002)')
+
+    .expect(pageModel.quantityInput.exists).ok()
+    .expect(pageModel.practiceExpandableSection.exists).ok()
+    .expect(await extractInnerText(pageModel.practiceExpandableSection))
+    .eql(content.solutionTable.cellInfo.quantity.expandableSection.title)
+    .expect(pageModel.practiceExpandableSection.find('details[open]').exists).notOk()
+    .click(pageModel.practiceExpandableSection.find('summary'))
+    .expect(pageModel.practiceExpandableSection.find('details[open]').exists).ok()
+    .expect(await extractInnerText(pageModel.practiceExpandableSection.find('.nhsuk-details__text')))
+    .eql(content.solutionTable.cellInfo.quantity.expandableSection.innerComponent.replace('<br><br>', ''))
+
+    .expect(pageModel.dateInput.exists).ok()
+    .expect(pageModel.dayInput.getAttribute('id')).eql('deliveryDate-day')
+    .expect(pageModel.dayInput.getAttribute('name')).eql('deliveryDate-day')
+    .expect(pageModel.dayInput.getAttribute('value')).eql('12')
+
+    .expect(pageModel.monthInput.getAttribute('id')).eql('deliveryDate-month')
+    .expect(pageModel.monthInput.getAttribute('name')).eql('deliveryDate-month')
+    .expect(pageModel.monthInput.getAttribute('value')).eql('10')
+
+    .expect(pageModel.yearInput.getAttribute('id')).eql('deliveryDate-year')
+    .expect(pageModel.yearInput.getAttribute('name')).eql('deliveryDate-year')
+    .expect(pageModel.yearInput.getAttribute('value')).eql('2020')
+
+    .expect(pageModel.dateExpandableSection.exists).ok()
+    .expect(await extractInnerText(pageModel.dateExpandableSection))
+    .eql(content.solutionTable.cellInfo.deliveryDate.expandableSection.title)
+    .expect(pageModel.dateExpandableSection.find('details[open]').exists).notOk()
+    .click(pageModel.dateExpandableSection.find('summary'))
+    .expect(pageModel.dateExpandableSection.find('details[open]').exists).ok()
+    .expect(await extractInnerText(pageModel.dateExpandableSection.find('.nhsuk-details__text')))
+    .eql(content.solutionTable.cellInfo.deliveryDate
+      .expandableSection.innerComponent.replace('<br><br>', ''));
 });
 
-test('should render the price with a value of 0 when returned from the API', async (t) => {
+test('should render empty price when 0 returned from the API', async (t) => {
   const mockSelectedPrice = { ...selectedPrice, price: 0 };
   await pageSetup({ ...defaultPageSetup, postRoute: true, mockData: mockSelectedPrice });
   await t.navigateTo(pageUrl);
-
-  const table = Selector('div[data-test-id="price-table"]');
-  const row = table.find('[data-test-id="table-row-0"]');
-  const priceInput = row.find('[data-test-id="question-price"] input');
+  const pageModel = new AdditionalServicePageModel();
 
   await t
-    .expect(priceInput.getAttribute('value')).eql(mockSelectedPrice.price.toString());
+    .expect(pageModel.priceInput.getAttribute('value')).eql(undefined);
 });
```

Comment:
I'm not sure this is correct.

If I remember correctly, £0.00 is considered a valid price that a supplier can offer so a zero-valued price should display as zero. However, it is probably worth double-checking with Lorraine to confirm.

Reply:
Checked and corrected